### PR TITLE
Bugfix/fix warmup scheduler setting error

### PR DESCRIPTION
**Pull request by zephyr-sh**
## ✨ Summary

This PR introduces a **polynomial LR scheduler with warmup** and enhances the **warmup scheduler infrastructure** with better modularity, validation, and test coverage.

#### 1. **Refactored `PolynomialLRWarmup`**

* Migrated to inherit from `torch.optim.lr_scheduler.LRScheduler` (modern PyTorch API).
* Rewritten with:

  * **Explicit closed-form logic** via `get_closed_form()`
  * Strict **input validation** for `warmup_iters`, `total_iters`, and `power`
  * Support for both **legacy API** (`step(epoch)`) and **modern chainable API**
* Behavior:

  * Linearly increases LR from 0 → base\_lr over `warmup_iters`
  * Then polynomial decay to 0 across remaining `total_iters - warmup_iters`
  * Fixed to **clamp** after `total_iters`

#### 2. **Introduced `WrappedLRScheduler`**

* A general-purpose warmup wrapper:

  * Handles linear warmup with configurable `multiplier`
  * Delegates to any other scheduler (`after_scheduler`) after warmup
* Works with any `LRScheduler` (e.g. `ReduceLROnPlateau`, `MultiStepLR`, etc.)
* Compatible with `step()` API (including metric-based schedulers)
* Includes strong argument checks and lifecycle safety (e.g. base LR re-init once)

#### 3. **Added `MultiStepLRWarmUp` factory function**

* Combines:

  * Linear warmup via `WrappedLRScheduler`
  * Followed by decay via `MultiStepLR`
* Clean and reusable for config-based instantiation

#### 4. **Registry Integration**

* Registered all schedulers (`PolynomialLRWarmup`, `WrappedLRScheduler`, `MultiStepLRWarmUp`) to `OPTIMIZERS` registry
* Updated `__init__.py` to reflect the new module inclusion

---

### `tests/base/optim/test_polynomial_lr_warmup.py`

* Validates:

  * Input validation
  * Correct LR output in warmup and decay phases
  * Behavior after `total_iters`
  * Compatibility with `.step()` chainable API

### `tests/base/optim/test_warm_up.py`

* Covers:

  * `WrappedLRScheduler` warmup-only behavior
  * Warmup w/ multiplier > 1
  * Correct delegation to `after_scheduler` (e.g. `MultiStepLR`)
  * Correct final LR holding behavior
  * Factory behavior and edge cases for `MultiStepLRWarmUp`

All tests pass.

---

## 🧩 Motivation

* PyTorch is progressively deprecating `_LRScheduler` in favor of `LRScheduler`.
* Polynomial warmup+decay is a common pattern not yet fully supported natively.
* Reusable warmup wrappers simplify composition and config loading.
* Provides a reliable warmup scheduling foundation for future models.

---

## 🔄 API Summary

```python
# Polynomial warmup + decay
PolynomialLRWarmup(optimizer, warmup_iters=5, total_iters=30, power=1.0)

# General warmup wrapper
WrappedLRScheduler(optimizer, milestone=5, multiplier=1.0, after_scheduler=...)

# Factory: warmup + MultiStep
MultiStepLRWarmUp(optimizer, milestones=[10, 20], warmup_milestone=3, gamma=0.1)
```

---

## 📎 Miscellaneous

* Ensures compatibility with both old and new scheduler API
* Fully documented with docstrings and param comments

---

## ✅ Checklist

* [x] Feature implemented with modular, typed design
* [x] Integrated with registry
* [x] Unit tests added
* [x] Passes all CI checks
* [x] Compatible with both legacy and chainable `.step()` API

### CODE REVIEW

**Review by kunkunlin1221 — APPROVED** [review]
LGTM
